CIPs/cip-0046.md

```diff
@@ -0,0 +1,128 @@
+* Creating an exchange proposal.
+* Gathering community feedback with the intent of having an exchange proposal approved.
+* Expected behavior of the approver.
+* Vetoing an exchange proposal.
+
+## Abstract
+Granda Mento, as described in [CIP 38](cip-0038.md), is a mechanism for large CELO <-> stable token exchanges through "exchange proposals" that anyone can create. If rough consensus from the community is achieved, the approver (a multisig of community member signers) is expected to approve an exchange proposal. After an exchange proposal is approved, a period of time exists where the community can veto the exchange proposal via a governance proposal. This document provides details on these processes.
+
+## Motivation
+While CIP 38 details the implementation of Granda Mento, there is no standard for the surrounding processes. Because Granda Mento relies upon community participation and approver multisig signers executing the rough consensus of the community, it's important for these processes to exist.
+
+## Specification
+
+### Creating a forum post
+
+To field opinions from the community and arrive at a rough consensus to approve an exchange proposal, an exchange proposal must create a post at https://forum.celo.org. A forum post must receive sufficient community input before the approver multisig can determine rough consensus.
+
+An on-chain exchange proposal does not need to be created when the forum post is created if the exchanger is hoping to gauge opinions before locking funds in GrandaMento. However, the exchange proposal should be submitted on-chain prior to achieving rough consensus to allow community members to verify the on-chain exchange proposal for themselves. 
+
+#### Forum post title format
+
+```
+GrandaMento Exchange Proposal <on-chain proposal ID> Discussion: <sell amount> <sell token> for <buy amount> <buy token>
+```
+For example: `GrandaMento Exchange Proposal 1 Discussion: 500,000 CELO for 1,500,000 cUSD`.
+
+If the proposal has not been created on-chain yet, the proposal ID and buy amount can be omitted for the time being, though it should be added later upon exchange proposal creation. For example: `GrandaMento Exchange Proposal <TBD> Discussion: 500,000 CELO for cUSD`.
+
+#### Forum post contents
+```
+1. What token and quantity are you selling, and what token are you buying?
+
+2. Who are you?
+
+3. Why do you want to execute the exchange?
+// Include the intended use of bought funds.
+
+4. What is the on-chain proposal ID and the on-chain buy amount?
+// Detailed info can be found by running this command and sharing the output:
+//   `celocli grandamento:show --proposalId <proposal-id>`
+// If you are planning to create the on-chain exchange proposal after
+// fielding opinions from the community, please mention so here.
+
+--- Only if selling CELO ---
+5. How did you acquire the CELO?
+// For example, via the Celo Foundation, Mento, a centralized exchange, OTC, etc.
+
+6. At what price did you acquire the CELO?
```

Comment:
I guess this question would make many players uncomfortable or goes against their internal privacy rules, can they alternatively disclosure the date of the purchase?

Reply:
Just added a note about exchangers being encouraged to answer all questions for transparency, and if they wish not to answer some they should explain why